### PR TITLE
Proposal for a new wallet UI

### PR DESCRIPTION
**Pull request by lorenz10-forker**
I have tried to simplify and modernize the UI, I'm not a designer so I did the best that I could.
I only changed these three screens for the moment. 
Let me know what you think.

Screenshot_1623765720
Screenshot_1623765757
Screenshot_1623765769
Screenshot_1623773482

### CODE REVIEW

**Comment by bananenwilly** [comment]
@Sentinelrv any comments on the color choices?

**Comment by lorenz10-forker** [comment]
The optimal situation that I imagined would have been having the list/send/receive tabs covering the four buttons when scrolling down, however I wasn't able figure it out.

**Comment by jointhepartypooper** [comment]
test for dark theme too as the colors are hard coded again...

**Comment by lorenz10-forker** [comment]
I'll try to fix the mess integrating my commits with the Material Theme tomorrow...

**Comment by lorenz10-forker** [comment]
I have tried to use more the app theme in the last commit.
The four blue buttons does not convince me totally, what do you think?

**Comment by Sentinelrv** [comment]
> @Sentinelrv any comments on the color choices?

Willy, I have not been able to test the new mobile wallet yet myself. I don't have Android and it only recently became available on iOS. I'll be sure to get it up and see what the differences are between the current version and this.

Generally as a rule though, I never place dark colors over top of any green background. It is always white or very light green text and icons over top of green backgrounds. Check our recently updated subreddit for an example of my color usage...

Light Mode: https://old.reddit.com/r/peercoin/
Dark Made: https://nm.reddit.com/r/peercoin/

**Comment by lorenz10-forker** [comment]
Screenshot_1623934690
Screenshot_1623934685
Screenshot_1623934874
Screenshot_1623934651
Screenshot_1623934645

UPDATE 

what has changed up to now:

-new UI for wallet home and wallet list (blue amount still does not convince but haven't found an alternative yet)
-address book moved into bottom bar (search address feature not yet updated, so is missing for the moment)
-removed app drawer, now paths are:

wallet list -> wallet
                -> settings (top left button) -> info (top right button)
                -> new wallet (top right button)

-purchase PPC moved under the receive widget
-share app moved in app settings

**Comment by marijacin** [comment]
hi! if you dont mind me commenting: 
1. why mixing in the dark blue colour, if its in no way related to Peercoin? Rather use dark grey. 
2. why put "address book" in bottom bar? in bottom bar should only go really important actions (send, receive, profile), that users will be using all the time. i feel like address book can easily go into "settings" or "profile"

**Comment by lorenz10-forker** [comment]
> hi! if you dont mind me commenting:
> 
>     1. why mixing in the dark blue colour, if its in no way related to Peercoin? Rather use dark grey.
> 
>     2. why put "address book" in bottom bar? in bottom bar should only go really important actions (send, receive, profile), that users will be using all the time. i feel like address book can easily go into "settings" or "profile"

1. I'll try other combinations
2. With Profile you mean transaction list or I should create a new tab called profile? And if it is the second what should I put in it?

**Comment by marijacin** [comment]
@lorenz10-forker maybe it would be smart to put "profile" instead of "transactions". That way, when you go to profile you can edit your addresses, see transactions, check address book, check your funds...

**Comment by lorenz10-forker** [comment]
[OFF TOPIC] I'd like to setup a server like has been done for this app, with Peercoind + ElectrumX. 
Is there a good tutorial guide for ElectrumX or did you follow the official documentation?

**Comment by peerchemist** [comment]
> [OFF TOPIC] I'd like to setup a server like has been done for this app, with Peercoind + ElectrumX.
> Is there a good tutorial guide for ElectrumX or did you follow the official documentation?

https://github.com/peercoin/electrumx-docker/

**Comment by lorenz10-forker** [comment]
> https://github.com/peercoin/electrumx-docker/

Thanks

**Comment by lorenz10-forker** [comment]
> > [OFF TOPIC] I'd like to setup a server like has been done for this app, with Peercoind + ElectrumX.
> > Is there a good tutorial guide for ElectrumX or did you follow the official documentation?
> 
> https://github.com/peercoin/electrumx-docker/

Thanks @peerchemist

**Comment by lorenz10-forker** [comment]
Screenshot_1624029578
Screenshot_1624029570
Screenshot_1624029574
Screenshot_1624029586

UPDATE 

I think those are better. Profile page still empty.
Next weeks I won't be able to work as much as I've done so far on the project.

**Comment by bananenwilly** [comment]
@lorenz10 I have been following your edits thoroughly and your latest changes look very good so far.
Please tell me when this is ready to review since I have already started twice and got outpaced by your changes 😊

**Comment by lorenz10-forker** [comment]
Regarding Send, Receive and Transactions tabs I'm almost done, unless there are other requests or suggestions, I think they are, at least, better than before (what you think?). 
The only doubt I have is whether they scale on different screen resolutions since I've used many boxes with fixed size to create them.

The profile page is empty for the moment, but I was thinking to place the following things (what you think?):
-a preview of user addresses (3/4) + their count + button to open theri list
-a preview of sending addresses (3/4) + their count + button to open their list
-the three options that currently are in the PopupMenu (three dots button top right for non developers) ?
-allowing to switch or create wallets here and skipping the first screen of the app?

In the next weeks changes will be slower than this week.

**Comment by bananenwilly** [comment]
couple of things I have noticed testing head ...624

- new wallet alert has no rounded corners as opposed to the rest of your design changes
- empty transaction list is just a green screen without any text 
- I did not manage to find the address book at this time
- Moving the three dot ... items to the profile tab is not a bad idea as such, for the previews you mentioned I don't see much of a benefit. I would also keep the wallet creation in the central home screen 
- Share App in Settings seems weird to me .. maybe bottom nav bar for the main screen? (wallet list) 
- receive tab: for testnet it should be "obtain .. testnet peercoin"  not "obtain peercoin" 
- dark mode is unusable at this point 

I have not tested this on tablets etc. yet, just on a pixel 4a.

**Comment by lorenz10-forker** [comment]
fixed
>     * [ ]  - new wallet alert has no rounded corners as opposed to the rest of your design changes
>     * [ ]  - empty transaction list is just a green screen without any text
>     * [ ]  - receive tab: for testnet it should be "obtain .. testnet peercoin"  not "obtain peercoin"
soon...
>     * [ ]  - I did not manage to find the address book at this time
I think its best concentrating all effort on a single good UI
>     * [ ]  - dark mode is unusable at this point
I ment smaller screens, on tablet I haven't tried anything
> I have not tested this on tablets etc. yet, just on a pixel 4a.

File: lib/widgets/receive_tab.dart [hunk 1]
```diff
@@ -83,129 +87,257 @@ class _ReceiveTabState extends State<ReceiveTab> {
...
-                    icon: Icon(Icons.money),
-                    labelText: AppLocalizations.instance
-                        .translate('receive_requested_amount'),
-                    suffix: Text(_wallet.letterCode),
+            ),
+            _wallet.title.contains('Testnet') ? PeerContainer(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: <Widget>[
+                  PeerServiceTitle(title: 'Obtain tPPC'),
```
Comment (bananenwilly) [inline lib/widgets/receive_tab.dart:297]: i18n

File: lib/widgets/receive_tab.dart [hunk 2]
```diff
@@ -83,129 +87,257 @@ class _ReceiveTabState extends State<ReceiveTab> {
...
-                label: Text(
-                  AppLocalizations.instance.translate('receive_share'),
-                ),
+                  Text(AppLocalizations.instance
+                      .translate('buy_peercoin_dialog_content'),
+                    textAlign: TextAlign.center,
+                  ),
+                  SizedBox(height: 20),
+                  PeerButton(
+                    text: 'Open faucet',
```
Comment (bananenwilly) [inline lib/widgets/receive_tab.dart:310]: i18n

File: lib/widgets/receive_tab.dart [hunk 3]
```diff
@@ -83,129 +87,257 @@ class _ReceiveTabState extends State<ReceiveTab> {
...
+                        .translate('buy_peercoin_dialog_title'),
+                    textAlign: TextAlign.center,
+                  ),
+                  Text(AppLocalizations.instance
+                      .translate('buy_peercoin_dialog_content'),
+                    textAlign: TextAlign.center,
+                  ),
+                  SizedBox(height: 20),
+                  PeerButton(
+                    text: 'Open website',
```
Comment (bananenwilly) [inline lib/widgets/receive_tab.dart:332]: i18n

File: lib/widgets/send_tab.dart [hunk 4]
```diff
@@ -50,6 +55,9 @@ class _SendTabState extends State<SendTab> {
       _availableAddresses =
           await _activeWallets.getWalletAddresses(_wallet.name);
       setState(() {
+        if(widget._passedAddress!=null){
```
Comment (bananenwilly) [inline lib/widgets/send_tab.dart:58]: please turn on the integrated flutter formater

File: lib/widgets/send_tab.dart [hunk 5]
```diff
@@ -249,172 +257,224 @@ class _SendTabState extends State<SendTab> {
...
-                }),
-            SizedBox(height: 30),
-            Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
-              ElevatedButton.icon(
-                style: ElevatedButton.styleFrom(
-                  primary: Theme.of(context).primaryColor,
+                          return null;
+                        }),
+                    SizedBox(height: 30),
+                    PeerButtonBorder(text: 'QR-Code',
```
Comment (bananenwilly) [inline lib/widgets/send_tab.dart:431]: i18n

File: lib/widgets/send_tab.dart [hunk 6]
```diff
@@ -249,172 +257,224 @@ class _SendTabState extends State<SendTab> {
...
+                        } else {
+                          ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                              content: Text(AppLocalizations.instance.translate(
+                                'send_errors_solve',
+                              ))));
+                        }
+                      },
+                    ),
+                    SizedBox(height: 10),
+                    Text('Hint: Scanning the QR-Code will save you some time',
```
Comment (bananenwilly) [inline lib/widgets/send_tab.dart:466]: i18n

File: lib/widgets/transactions_list.dart [hunk 7]
```diff
@@ -47,161 +49,216 @@ class _TransactionListState extends State<TransactionList> {
...
-                          child: Text(
-                            resolveAddressDisplayName(_filteredTx[i].address),
-                            overflow: TextOverflow.ellipsis,
-                            textScaleFactor: 1,
-                          ),
-                        ),
-                        trailing: Column(
-                          mainAxisAlignment: MainAxisAlignment.center,
+        _filteredTx==null || _filteredTx.isEmpty? Center(
+            child: Text('Empty list',
```
Comment (bananenwilly) [inline lib/widgets/transactions_list.dart:101]: i18n

Comment (lorenz10-forker) [reply lib/widgets/transactions_list.dart:101]: What does i18n means?

Comment (bananenwilly) [reply lib/widgets/transactions_list.dart:101]: Translation

**Comment by lorenz10-forker** [comment]
Screenshot_1624451282
Screenshot_1624451865

UPDATE
- "share app" opens if is pressed the Peercoin icon in wallet_list, I wasn't able to add some splash effect on tap
- addresses UI does not totally convince me, for the moment addresses are in the bottom bar but can be easily moved up in top bar

**Comment by bananenwilly** [comment]
thanks! will give it a deep review tomorrow

**Comment by bananenwilly** [comment]
@lorenz10 I have made some major changes in the meantime and those conflicts above would need resolving before I can continue to review

**Comment by lorenz10-forker** [comment]
> @lorenz10 I have made some major changes in the meantime and those conflicts below would need resolving before I can continue to review

@bananenwilly ok I will fix those conflicts as soon as possible, in exchange I have a question to ask you about my private Peercoin fork, can I write it to you on Telegram ?

**Comment by bananenwilly** [comment]
Sure

**Comment by bananenwilly** [comment]
closed on request of @lorenz10

**Pull request closed without merging**